src/crypto/tls/handshake_client.go

```diff
@@ -586,6 +586,12 @@ func (hs *clientHandshakeState) doFullHandshake() error {
 		if err != nil {
 			return err
 		}
+
```

Comment:
Should we move this before the above `c.readHandshake()`?

Reply:
Done.